src/transformers/trainer.py

```diff
@@ -156,10 +158,15 @@
 if is_apex_available():
     from apex import amp
 
+if is_ipex_available() and "--use_ipex" in sys.argv:
+    import intel_extension_for_pytorch as ipex
```

Comment:
We can't rely on `sys.argv` as the arguments may be passed to the `Trainer` class as `TrainingArguments`. This import should be done inside the `Trainer` as needed.

Reply:
Got it, and have moved this importation to the inside `ipex_optimize_model`